src/runtime_src/core/include/xclbin.h

```diff
@@ -455,6 +455,30 @@ extern "C" {
     };
     XCLBIN_STATIC_ASSERT(sizeof(struct clock_freq_topology) == 138, "clock_freq_topology structure no longer is 138 bytes in size");
 
+    enum XRT_RESOURCE_QUERY_TYPE {
+        IPU_CLK_MAX,
+        IPU_TOPS_MAX,
+        IPU_TASK_MAX,
+        IPU_TOPS_CURR,
+        IPU_TASK_CURR
+    };
+
+    struct res_entry {                     /* Resource Info Entry */
+        uint8_t m_type;                    /* Resource type (enum XRT_RESOURCE_QUERY_TYPE) */
+        uint8_t m_unused[7];               /* Not used - padding */
+        union {
+            uint64_t data_uint64;          /* Member for integer value */
+            double data_double;            /* Number for double value */
+        };
+    };
+    XCLBIN_STATIC_ASSERT(sizeof(struct res_entry) == 16, "res_entry structure no longer is 16 bytes in size");
+
+    struct res_info {                      /* Resource Infor section */
+        uint16_t m_count;                  /* Number of entries */
+        struct res_entry m_res[1];         /* resource info data */
+    };
+    XCLBIN_STATIC_ASSERT(sizeof(struct res_info) == 24, "res_info structure no longer is 24 bytes in size");
```

Comment:
I just followed the example of clock topology. This is to add TOPs number to the platform report in xrt-smi tool. Maybe xclbin.h is not a good place for these structures. could you suggest the proper header file to host this so that both xrt-smi and kmd driver can reference them? thanks.

Reply:
Right, xclbin is not the proper place.  @AShivangi can suggest.

Reply:
fixed.